lib/rust/shapely/macros/src/tagged_enum.rs

```diff
@@ -0,0 +1,298 @@
+use crate::prelude::*;
+
+use inflector::cases::snakecase::to_snake_case;
+use syn::Data;
+use syn::DeriveInput;
+use syn::Fields;
+
+
+
+// ===================
+// === Entry Point ===
+// ===================
+
+/// Transforms Rust enums into enums where each variant is a separate type. It also implements
+/// several traits (such as conversions between variants and the enum type) and defines utility
+/// functions, such as constructors.
+///
+/// To learn more about what code is being generated, parts of the code generation were provided
+/// with comments showing the output of application of this macro to the following structure:
```

Comment:
I'd expect docstring to provide at least information:
* what traits are implemented
* what definitions (names) are introduced for user

Reply:
Hmm, but this is provided if you scroll the code down. I'm not sure if copy-pasting these code fragments here would make it more readable (I think it would destroy the redability). However, maybe it can be improved. I'd love to have short call about it.